packages/eslint-plugin/src/rules/prefer-readonly.ts

```diff
@@ -251,6 +251,13 @@ type ParameterOrPropertyDeclaration =
 const OUTSIDE_CONSTRUCTOR = -1;
 const DIRECTLY_INSIDE_CONSTRUCTOR = 0;
 
+enum TypeToClassRelation {
```

Comment:
[Praise] I found this enum and the logic around it to be quite helpful in understanding the new logic. Very readable. Nice! 👏